### PR TITLE
Feature/as 4143 eligibility journey flow

### PR DESCRIPTION
**Pull request by shilpabasu22**
## Ticket Number

https://pins-ds.atlassian.net/browse/AS-4143

## Description of change
Complete Eligibility journey tests

## Checklist

- [ ] Requires infrastructure changes
- [ ] If adding or remove environment variables (e.g. in `docker-compose.yaml`) then I have updated the appropriate Helm chart
- [ ] I have updated the documentation accordingly
- [ ] My commit history in this PR is linear
- [ ] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `master` (please only [rebase](https://github.com/foundry4/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

File: packages/e2e-tests/cypress/integration/eligibility/householder-planning/complete-eligibility-journey/complete-eligibility-journey.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+} from "../../../../support/eligibility/date-decision-received/enter-date-householder-decision-received";
+import {
+  enterDateDecisionDueHouseholder
+} from "../../../../support/eligibility/date-decision-due-householder/enter-date-decision-due-householder";
+import {getClaimingCostNo} from "../../../../support/eligibility/page-objects/claiming-costs-po";
+const url = 'before-you-start/local-planning-depart';
+before(()=>{
+  cy.task('getData',{collection:'appeals',filter:"appeal.state"});
+});
+Given('appellant selects local planning department',()=>{
```
Comment (nssdesikan) [inline packages/e2e-tests/cypress/integration/eligibility/householder-planning/complete-eligibility-journey/complete-eligibility-journey.js:34]: New line please

File: packages/e2e-tests/cypress/integration/eligibility/householder-planning/complete-eligibility-journey/complete-eligibility-journey.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  verifyPageHeading('What type of planning application is your appeal about?');
+  selectPlanningApplicationType(applicationType);
+});
+
+Given('appellant selects the option as No for listed building',()=>{
+  verifyPageTitle('Is your appeal about a listed building? - Before you start - Appeal a planning decision - GOV.UK');
+  verifyPageHeading('Is your appeal about a listed building?');
+  selectListedBuildingDecision('No');
+});
+Given('appellant selects the {string}',(application_decision)=>{
```
Comment (nssdesikan) [inline packages/e2e-tests/cypress/integration/eligibility/householder-planning/complete-eligibility-journey/complete-eligibility-journey.js:57]: New line please

**Review by nssdesikan — COMMENTED** [review]
Anything is going to come up in the future for the following:-
1. End to End Back link
2. User moving from one appeal type to other?

**Comment by shilpabasu22** [comment]
> Anything is going to come up in the future for the following:-
> 
> 1. End to End Back link
> 2. User moving from one appeal type to other?

That should be in a different ticket (1. I will cover with back button ticket, 2.  will be covered with the new ticket when the journey is more clear between various appeal types)

File: packages/e2e-tests/cypress/integration/eligibility/full-appeal/complete-eligibility-journey/complete-eligibility-journey.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+} from "../../../../support/eligibility/date-decision-received/enter-date-decision-received";
+import {getDate, getMonth, getYear} from "date-fns";
+import {enterDateDecisionDue} from "../../../../support/eligibility/date-decision-due/enter-date-decision-due";
+import {getEnforcementNoticeNo} from "../../../../support/eligibility/page-objects/enforcement-notice-po";
+import {verifyPage} from "../../../../support/common/verifyPage";
+const url = 'before-you-start/local-planning-depart';
+before(()=>{
+  cy.task('getData',{collection:'appeals',filter:"appeal.state"});
+});
+Given('appellant selects local planning department',()=>{
```
[the remark this answers is not in the document]
Comment (shilpabasu22) [reply packages/e2e-tests/cypress/integration/eligibility/full-appeal/complete-eligibility-journey/complete-eligibility-journey.js:32]: updated as per review comments

File: packages/e2e-tests/cypress/integration/eligibility/full-appeal/complete-eligibility-journey/complete-eligibility-journey.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+  verifyPageHeading('What type of planning application is your appeal about?');
+  selectPlanningApplicationType(applicationType);
+});
+
+Given('appellant selects {string} from the list of options',(option)=>{
+  verifyPageTitle('Was your planning application about any of the following? - Before you start - Appeal a planning decision - GOV.UK');
+  verifyPageHeading('Was your planning application was about any of the following?');
+  selectSiteOption(option);
+});
+Given('appellant selects the {string}',(application_decision)=>{
```
[the remark this answers is not in the document]
Comment (shilpabasu22) [reply packages/e2e-tests/cypress/integration/eligibility/full-appeal/complete-eligibility-journey/complete-eligibility-journey.js:56]: updated as per review comments

File: packages/e2e-tests/cypress/integration/eligibility/householder-planning/complete-eligibility-journey/complete-eligibility-journey.js (hunk 2 above)
Comment (shilpabasu22) [reply packages/e2e-tests/cypress/integration/eligibility/householder-planning/complete-eligibility-journey/complete-eligibility-journey.js:60]: updated as per review comments

File: packages/e2e-tests/cypress/integration/eligibility/householder-planning/complete-eligibility-journey/complete-eligibility-journey.js (hunk 1 above)
Comment (shilpabasu22) [reply packages/e2e-tests/cypress/integration/eligibility/householder-planning/complete-eligibility-journey/complete-eligibility-journey.js:36]: updated as per review comments

**Review by nssdesikan — APPROVED** [review]
LGTM

**Review by rcmsoftwaredevelopment — DISMISSED** [review]

**Pull request closed without merging**